contracts/handler/CompatibilityFallbackHandler.sol

```diff
@@ -150,7 +150,7 @@ contract CompatibilityFallbackHandler is TokenCallbackHandler, ISignatureValidat
             mstore(0x40, add(response, responseSize))
             returndatacopy(response, 0x20, responseSize)
 
-            if iszero(mload(0x00)) {
+            if iszero(mload(add(ptr, calldatasize()))) {
```

Comment:
This can come from the scratch space :point_up:.

Also, this is UB - if the `returndatasize` is greater than the `calldatasize`, you will overwrite what you are trying to read here.

Reply:
Just to be clear, as I didn't do:
`mstore(0x40, add(calldatasize(), 0x20))` before assigning the free memory pointer for `response`, the boolean result of the `call` can get overwritten if the `returndatasize` is greater than `calldatasize`.

Reply:
Correct, this is why the first 32 bytes of the response data are written and read from the scratch space.

Reply:
Got it, thanks!

Made the change 👍🏾